app/javascript/controllers/status_controller.js

```diff
@@ -9,17 +9,40 @@ export default class extends Controller {
     }
   }
 
+  disconnect() {
+    this.stopRefreshing()
+  }
+
   startRefreshing() {
-    setInterval(() => {
+    this.interval = setInterval(() => {
       this.load()
     }, this.data.get("refreshInterval"))
   }
 
+  stopRefreshing() {
+    if(this.interval) {
+      clearInterval(this.interval)
+    }
+  }
+
   load() {
     fetch(this.data.get("url"))
-    .then(response => response.text())
-    .then(html => {
-      this.element.innerHTML = html
-    })
+      .then(response => response.text())
```

Comment:
why all the `.then`s, and not a single method that does all the things? I'm not familiar with the fetch/then pattern, so I'm probably missing something obvious.

Reply:
This is a [Promise chain](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises) which I like to break up into atomic units to make it easy to recover from errors (for example if JSON parsing fails)